src/functions-and-arrays.js

```diff
@@ -1,41 +1,89 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(firstNumber, secondNumber) {
+
+  if (firstNumber < secondNumber) {
+    return secondNumber
+  }
+  else {
+    return firstNumber
+  }
+}
+
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null
+  }
+  let longestWord = ''
+  for (let i = 0; i < words.length; i++); {
+    if (words[i].length > word.length) {
+
+    }
+  }
+  return longestWord
+}
 
 
 // Iteration #3: Calculate the sum
-const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10]
+
+function sumNumbers(numbers) {
+  let total = 0
+  for (let i = 0; i < numbers.length; i++) {
+    total += numbers[i]
+  }
+  return total
+}
 
-function sumNumbers() {}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum() { }
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbers) {
+  if (numbers.length === 0) {
+    return null
+  }
+  let media = 0
+  media = sumNumbers(numbers) / numbers.length
+  return media
+}
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsAvg) {
+  if (wordsAvg.length === 0) {
+    return null
+  }
+  let mediaWords = 0
+  let totalLength = 0
+  for (let i = 0; i < wordsAvg[i].length; i++) {
```

Comment:
Tienes dos pequeños fallos, al recorrer el array con el bucle for no le metes el indice al array.
Y se te olvidó retornar el valor pedido al final de la función:

```javascript
for (let i = 0; i < wordsAvg.length; i++) {
    totalLength += wordsAvg[i].length
  }
  mediaWords = totalLength / wordsAvg.length
  return mediaWords
```